chain/chain/src/flat_storage_creator.rs

```diff
@@ -361,39 +362,45 @@ impl FlatStorageShardCreator {
 
 /// Creates flat storages for all shards.
 pub struct FlatStorageCreator {
-    pub shard_creators: Vec<FlatStorageShardCreator>,
+    pub shard_creators: HashMap<ShardId, FlatStorageShardCreator>,
     /// Used to spawn threads for traversing state parts.
     pub pool: rayon::ThreadPool,
 }
 
 impl FlatStorageCreator {
     pub fn new(
```

Comment:
Add comments for this function mentioning that it either creates FlatStorage or starts the migration.